core/src/main/java/org/stellar/anchor/event/EventService.java

```diff
@@ -1,14 +1,94 @@
 package org.stellar.anchor.event;
 
+import java.util.List;
 import org.stellar.anchor.api.event.AnchorEvent;
+import org.stellar.anchor.api.exception.AnchorException;
 import org.stellar.anchor.api.exception.EventPublishException;
-import org.stellar.anchor.sep24.Sep24Transaction;
-import org.stellar.anchor.sep31.Sep31Transaction;
 
+/**
+ * The EventService is used to publish events to the event queue and to read events from the event
+ * queue.
+ *
+ * <p>Read event example
+ *
+ * <pre>
+ *     EventService eventService = new EventServiceImpl();
+ *     eventService.publish(new TransactionEvent(...));
+ *     eventService.publish(new ControlEvent(...));
+ *     ...
+ *     EventService.Session session = eventService.createSession(EventService.EventQueue.TRANSACTION);
+ *     EventService.ReadResponse readResponse = session.read();
+ *     List&lt;AnchorEvent&gt; events = readResponse.getEvents();
+ *     ...
+ *     session.ack(readResponse);
+ *     ...
+ *     session.close();
+ *     ...
+ *     eventService = null;
+ * </pre>
+ *
+ * <p>Publish event example
+ *
+ * <pre>
+ *     EventService eventService = new EventServiceImpl();
+ *     EventService.Session session = eventService.createSession(EventService.EventQueue.TRANSACTION);
+ *     session.publish(new TransactionEvent(...));
+ *     session.close();
+ * </pre>
+ */
 public interface EventService {
-  void publish(Sep24Transaction txn, AnchorEvent.Type type);
+  /**
+   * To be deprecated. Use createSession() instead.
+   *
+   * @param event the event to be published
+   * @throws EventPublishException
+   */
+  void publish(AnchorEvent event) throws EventPublishException;
 
-  void publish(Sep31Transaction txn, AnchorEvent.Type type);
+  /**
+   * Creates a session for publishing and reading events.
+   *
+   * @return a session object.
+   */
+  Session createSession(EventQueue eventQueue);
 
-  void publish(AnchorEvent event) throws EventPublishException;
+  interface Session {
+    /**
+     * Publishes an event to the event queue. The queue will be determined by the implementation of
+     * the Session.
+     *
+     * @param event the event to publish
+     */
+    void publish(AnchorEvent event) throws AnchorException;
+
+    /**
+     * Reads events from the event queue.
+     *
+     * @return
+     */
+    ReadResponse read() throws AnchorException;
+
+    /**
+     * Acknowledges that the events returned by the read() method have been processed.
+     *
+     * @param readResponse
+     */
+    void ack(ReadResponse readResponse) throws AnchorException;
```

Comment:
Does this map to a Kafka consumer ack, and DeleteMessageRequest in SQS?

Reply:
A SQS `ReceiveMessage` response can contain up to 10 messages. If any of those messages fail to process, we either ack all, or ack none. Should we have per message acks instead, or introduce a batch ack API?

Reply:
> Does this map to a Kafka consumer ack, and DeleteMessageRequest in SQS?
I think this can be delegated to the implementation. In Kafka, it would be `commit` and in SQS, it would be `DeleteMessageRequest`. I estimate both Kafka and SQS will be able to take care of the `ack()` requirement.

> A SQS `ReceiveMessage` response can contain up to 10 messages. If any of those messages fail to process, we either ack all, or ack none. Should we have per message acks instead, or introduce a batch ack API?

This is a good point. However, per-message ack is not possible in Kafka. In this scenario, if the server is down, all messages will not be committed. If some of the messages cannot be processed, they will be sent to a DLQ. The rest will be processed. And all messages will be acked.